services/venice-server/src/test/java/com/linkedin/venice/listener/StorageReadRequestHandlerTest.java

```diff
@@ -318,6 +414,22 @@ public void testMultiGetNotUsingKeyBytes(Boolean isParallel) throws Exception {
     for (int i = 0; i < recordCount; i++) {
       assertEquals(results.get(i), allValueStrings.get(i));
     }
+
+    /**
+     * The assertions below can catch an issue where metrics are inaccurate during parallel batch gets. This was due to
+     * a race condition which is now fixed. Unfortunately, the race is hard to hit, even on the buggy code, so the test
+     * succeeding is not a guarantee of the absence of a regression (but the test failing certainly indicates that).
+     *
+     * We are making the test fail reliably (in the presence of buggy stats handling) via a backdoor available through
+     * {@link ReadResponseLatencyInjector}.
```

Comment:
neat

Reply:
I don't like it a lot to be honest, but I think the overhead should be negligible and it's good to have peace of mind 🤷 ...

Reply:
Thanks for the review on PR 1152. I merged it yesterday, but this testing backdoor I left behind to reliably repro the race condition did not sit well with me... I took another stab at it here:

https://github.com/linkedin/venice/pull/1161

And I also threw in a few other cleanups that I had pulled out of the previous PR because that one was already big and I wanted to reduce its scope.